miden-tx/src/compiler/mod.rs

```diff
@@ -167,7 +158,8 @@ impl TransactionComplier {
         }
 
         // Create the [AssemblyContext] for compilation of the transaction program
-        let mut assembly_context = AssemblyContext::new(AssemblyContextType::Program);
+        let mut assembly_context =
+            AssemblyContext::for_program(&ProgramAst::parse("begin end").unwrap());
```

Comment:
This feels a bit hacky to me. The problem will go away once we migrate to dynamic note script invocations - but I think, maybe a better way would be to make `ProgramAst` an optional parameter for `for_program()` constructor. I'll make a PR in Miden VM to do this.

Reply:
Opened the PR here: https://github.com/0xPolygonMiden/miden-vm/pull/1070

Reply:
I agree that this feels hacky.  Why will this go away when we migrate to dynamic note script invocations (I'm not sure that it will)?

Reply:
We'll have a stable AST for the transaction program then. So, we'll be able to instantiate the context using this AST.

Reply:
At the moment we build the transaction program by hand, as such we don't have a AST that fully represents a program.  I guess once we migrate to dynamic note invocations we wont need to build the transaction program by hand anymore, we should be able to define the full program using masm and as such have a stable and complete AST for the transaction program.  Is this your thinking @bobbinth ?

Reply:
Yes - exactly!